hazelcast/src/test/java/com/hazelcast/jet/core/ObservableRingbufferContainerLeakTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.core;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.internal.services.ObjectNamespace;
+import com.hazelcast.jet.JetService;
+import com.hazelcast.jet.Observable;
+import com.hazelcast.jet.impl.execution.DoneItem;
+import com.hazelcast.ringbuffer.OverflowPolicy;
+import com.hazelcast.ringbuffer.Ringbuffer;
+import com.hazelcast.ringbuffer.impl.RingbufferContainer;
+import com.hazelcast.ringbuffer.impl.RingbufferService;
+import com.hazelcast.test.Accessors;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.NightlyTest;
+import com.hazelcast.test.annotation.Repeat;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.Iterator;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(NightlyTest.class)
+public class ObservableRingbufferContainerLeakTest extends JetTestSupport {
+
+    /**
+     * Regression test for #19580
+     */
+    @Test
+    @Repeat(100)
```

Comment:
leftover repeat?

Reply:
No this is here because the test almost never fails on the first try because of the race conditions. Even with the repeats the test takes around 5 secs.

Reply:
No, the test doesn't reproduce it 100 %, so I added the repeat. 100 was quite reliable on my machine.

It is marked `NightlyTest` not to interfere with the PR builder time (although it's just few seconds).